Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
IMO you can consider this a second bug, then. Taking your last statement:
> Requesting the size of a supplementary view at section 1 should return CGSizeZero b/c the object is out of bounds.
If you create a test for that in the current codebase, it will fail, as it won't return CGSizeZero, instead it will return the CGSize that you have defined for the IGListSupplementaryViewDelegate. But that is not a concern to me. The method's responsibility is to return the size for a supplementary view at a given index path, if you want to add the responsibility of checking if the index path actually exists, that's cool. In my test, I'm considering such responsibility doesn't exist, so the method is just returning the size for a section with index = 1. It wouldn't be able to do so in the current code base as the only possible index is 0.

I see you won't be convinced, which should mean I'm wrong, so please instruct me which is the expected way to return different sizes for different section indexes. Thanks @rnystrom !